File: Inverted-Index/src/file_handler/DirectoryReader.java
package file_handler;

import java.io.File;

import exceptions.NotDirectoryException;

public class DirectoryReader implements IDirectoryReader {

    @Override
    public File[] getFiles(String dirPath) throws NotDirectoryException {
        var path = new File(dirPath);
        if(path.isDirectory()) {
            return path.listFiles();
        }
        throw new NotDirectoryException(dirPath);
    }
    
}

File: Phase5-TDD/InvertedIndex/FileHandler/DirectoryReader.cs
using System;
using System.IO;
using InvertedIndex.Exceptions;

namespace InvertedIndex.FileHandler
{
    public class DirectoryReader : IDirectoryReader
    {
        public FileInfo[] GetFiles(string dirPath)
        {
            var path = new DirectoryInfo(dirPath);
            if(path.Exists) {
                return path.GetFiles();
            }
            throw new NotDirectoryException(dirPath);
        }
    }
}